### PR TITLE
Orca demo

### PR DESCRIPTION
**Pull request by hkvision**
https://calvinfeng.gitbook.io/machine-learning-notebook/supervised-learning/recommender/neural_collaborative_filtering

### CODE REVIEW

**Comment by jason-dai** [comment]
Is this similar to https://bigdl.readthedocs.io/en/latest/doc/UseCase/xshards-pandas.html and https://bigdl.readthedocs.io/en/latest/doc/UseCase/spark-dataframe.html?

**Comment by hkvision** [comment]
> Is this similar to https://bigdl.readthedocs.io/en/latest/doc/UseCase/xshards-pandas.html and https://bigdl.readthedocs.io/en/latest/doc/UseCase/spark-dataframe.html?

Yes, actually a combination of the two, using the xshards data loading and preprocessing from the former and the keras model definition from the latter. And changed the metrics and loss to mean_squared_error.

**Comment by hkvision** [comment]
Conda environment on driver:
- pip install https://sourceforge.net/projects/analytics-zoo/files/dllib-py-spark3/bigdl_dllib_spark3-0.14.0b20211123-py3-none-manylinux1_x86_64.whl 
- pip install https://sourceforge.net/projects/analytics-zoo/files/orca-py-spark3/bigdl_orca_spark3-0.14.0b20211123-py3-none-manylinux1_x86_64.whl
- pip install tensorflow==1.15.0
- pip install sklearn
- pip install pandas

**Comment by hkvision** [comment]
Steps:
```
1. Go to almaren gateway
2. Go to Almaren-Node-188
3. source activate orca-demo
4. `export PYSPARK_PYTHON=/usr/local/envs/pytf1/bin/python`
5. cd /opt/work/kai
6. python ncf_orca.py
```

**Comment by hkvision** [comment]
https://grouplens.org/datasets/movielens/

**Comment by hkvision** [comment]
ml-25m has 2000w records for ratings.
ml-latest-small has 8w records for ratings.
They should be good enough for the demo as large and small datasets.
ml-1m ratings is dat file with :: as separator; for 25m and latest-small, they are standard csv files.

**Comment by hkvision** [comment]
For TF1 Estimator, if metrics is not specified, it wouldn't do evaluation even if validation data is specified.
For keras, if metrics is not specified, it would evaluate using loss.
